Marlin/Marlin_main.cpp

```diff
@@ -1287,7 +1288,25 @@ static void update_software_endstops(AxisEnum axis) {
   {
     sw_endstop_min[axis] = base_min_pos(axis) + offs;
     sw_endstop_max[axis] = base_max_pos(axis) + offs;
+
+    #if ENABLED(AUTO_BED_LEVELING_FEATURE)
+      if (axis == Z_AXIS) {
```

Comment:
Did you see the asymmetry between https://github.com/MarlinFirmware/Marlin/pull/3829/files#diff-1cb08de130a6ece2d1b5b9c37bcfef48R1393-1403
```cpp
#if ENABLED(AUTO_BED_LEVELING_FEATURE) && Z_HOME_DIR < 0
```
in `set_axis_is_at_home()` and here? Does that make a difference in the resulting values? Sorry, too tired to analyse now.